src/box/lua/experimental/connpool.lua

```diff
@@ -0,0 +1,191 @@
+local config = require('config')
+local checks = require('checks')
+local netbox = require('net.box')
+
+local connections = {}
```

Comment:
Is there any mechanism to clean this table? It is possible to overflow it somehow during long runtime?

Reply:
At the moment the only way to clear the table is to restart Tarantool. As for overflow, we have standard Lua problems here. What I mean is that if you're using a GC64, you have a maximum of 2^26 keys, otherwise you'll likely hit the 2GB memory limit sooner.

Reply:
The main problem here that you can silently get overflow error and it will be very hard to understand exact reason.

Reply:
I don't think that we can realistically reach any Lua memory/Lua table size limit here. However, I agree that it would be good to delete items that are not in the config anymore. We can do it in `connect()`.

I wouldn't insist on it in the scope of this PR.